7/Constructor1/script.js

```diff
@@ -0,0 +1,69 @@
+"use strict";
+
+let mainTextAddY = document.getElementById("mainTextAddY");
+let mainTextAddN = document.getElementById("mainTextAddN");
+let mainText = document.getElementById("mainText");
+
+mainTextAddY.onclick = function(){
+    mainText.removeAttribute('disabled');
+};
+
+mainTextAddN.onclick = function(){
+    mainText.setAttribute('disabled', 'disabled');
+    mainText.style.opacity(0.5);
+};
+
+function getData(){
+    let newHTML;
+    //приём данных из формы
+    let title = title.value;// название
+    let description = description.value;// слоган
+    let menu = menu.value;// меню навигации
+    let LogoImg = logoImg.value;// логотип
+    let mainText = mainText.value;// основной текст
+    let colorTheme = colorTheme.value;// цветовая тема
+    let colorText = colorText.value;// цвет текста
+
+    document.body.style.backgroundColor = colorTheme;
+
+    //cоздание итоговой строки
+    newHTML = '<div id="main"><header><img src="' + LogoImg + '" alt="logo">';
+    newHTML += '<h1>' + title + '</h1>';
+    newHTML += '<h2>' + description + '</h2></header>';
+
+    //создание строки пунктов меню
+    let list = document.constructor.list;
+    let listItems = [];
+    let li = '';
+    
+    for (var i = 0; list[i]; ++i){
+        if(list[i].checked){
+            listItems[i] = list[i].value;
+            li += '<li><a href = "#">' + listItems[i] + ' ' + '</a></li>';
+        };
```

Comment:
please try to use ```Array.map```: 
```javascript
list.map( item => {
   if(item.checked) {
    ....
   }
})
```

Reply:
I changed the code here in more simple way